src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -376,10 +371,16 @@ public void RemoveSubscriber_Impl(GuidId subscriptionId, StreamId streamId)
             catch (Exception exc)
             {
                 receiverInitTask = null;
-                logger.Error(ErrorCode.PersistentStreamPullingAgent_12, $"Exception while trying to read from queue {queueId}", exc);
+                logger.Error(ErrorCode.PersistentStreamPullingAgent_12, $"Giving up reading from queue {queueId} after retry attempts {ReadLoopRetryMax}", exc);
             }
         }
-        
+
+        private bool ReadLoopRetryExceptionFilter(Exception e, int retryCounter, QueueId queueId)
```

Comment:
QueueId is a member variable, correct?  No need to pass it in.  If removed from signature, it would remove the need of  a lambda to call this.